src/Infrastructure.VS/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -31,13 +31,20 @@ namespace SonarLint.VisualStudio.IssueVisualization.Editor.LanguageDetection
     public interface ISonarLanguageRecognizer
     {
         IEnumerable<AnalysisLanguage> Detect(string filePath, IContentType bufferContentType);
+
+        AnalysisLanguage? GetAnalysisLanguageFromExtension(string extension);
```

Comment:
Parameter name: we're now expecting the `fileName` rather than the `extension`, aren't we?